strawberryfields/ops.py

```diff
@@ -569,15 +569,14 @@ def apply(self, reg, backend, **kwargs):
         """
         values = super().apply(reg, backend, **kwargs)
         # convert the returned values into an iterable with the measured modes indexed along
-        # the first axis and shots along second axis (if larger than 1), so that we can assign
+        # the first axis and shots along second axis, so that we can assign
         # register values
         shots = kwargs.get("shots", 1)
         if self.ns == 1:
             values = [values]  # values is either a scalar, or has shape (shots,)
         else:
-            if shots > 1:
-                values = values.T  # shape of values would be (shots, num_meas,)
-
+            if shots <= 0:
+                raise ValueError("Number of shots must be greater than zero.")
```

Comment:
`Measurement.apply()` does not need to know about shots, it just needs to slice the 2d or 3d array it receives into the RegRefs. It's good to test that shots > 0, but it should happen already in `BaseEngine._run()`.